server/src/main/kotlin/io/spine/examples/pingh/server/LocalServer.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.pingh.server
+
+import io.ktor.client.engine.cio.CIO
+import io.spine.environment.DefaultMode
+import io.spine.examples.pingh.clock.IntervalClock
+import io.spine.examples.pingh.github.ClientId
+import io.spine.examples.pingh.github.ClientSecret
+import io.spine.examples.pingh.github.GitHubApp
+import io.spine.examples.pingh.github.of
+import io.spine.examples.pingh.mentions.RemoteGitHubSearch
+import io.spine.examples.pingh.mentions.newMentionsContext
+import io.spine.examples.pingh.sessions.RemoteGitHubAuthentication
+import io.spine.examples.pingh.sessions.RemoteGitHubUsers
+import io.spine.examples.pingh.sessions.newSessionsContext
+import io.spine.server.Server
+import io.spine.server.ServerEnvironment
+import io.spine.server.delivery.Delivery
+import io.spine.server.storage.memory.InMemoryStorageFactory
+import io.spine.server.transport.memory.InMemoryTransportFactory
+import java.util.Properties
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * The port on which the Pingh server runs.
+ */
+private const val pinghPort = 50051
```

Comment:
We already have this constant defined in `Application`.

And it's also unnatural that `LocalServer` does not use `Application`, but rather copies some pieces of logic from it.

Let's re-do the whole "local" thing from scratch. 

1. Have a single `Server.kt`.
2. With `Application`, have two factory methods:
     * `local(...)`, which would create an instance of `Application` configured (along with the server environment) for the local launch mode;
     * `cloud(...)`, which would do the same for the Cloud-based run.
3. In `Server.kt` (which would remain the only "...Server" routine), detect whether we are in GCE or not. And call the corresponding factory method.

The easiest way to implement p.3 would be to have a system-level variable, which if set, would mean the cloud environment. Please also check whether there is such a system variable available in GCE environment already, provided by default.

The following code should also take that into account. 
```
ServerEnvironment.`when`(DefaultMode::class.java)
...
```
As the matter of fact, you could do the whole configuration rely on the detected type of `ServerEnvironment`. To do so, you should introduce your own type of the environment, and make it say "yes, it's me who's active" basing on the same mechanism described above (i.e. system variable).

Please think this whole comment over. And if needed, prepare to ask questions.